ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -380,4 +374,87 @@ public Optional<Boolean> doesAddressExist(
 
     return Optional.of(worldState.get(address) != null);
   }
+
+  @FunctionalInterface
+  public interface SimulationGasLimitCalculator {
+    long calculate(final long txGasLimit, final long blockGasLimit);
+
+    static long bounded(final long txGasLimit, final long blockGasLimit, final long rpcGasCap) {
```

Comment:
I think if we replace 0 values with `Long.MAX_VALUE` the logic would be as simple as taking a min over 3 values. Would it be simpler to read?